src/aleph/web/controllers/storage.py

```diff
@@ -177,22 +289,17 @@ async def _check_and_add_file(
             address=message_content.address,
             size=file.size,
         )
-
     else:
         message_content = None
 
-    # TODO: this can still reach 1 GiB in memory. We should look into streaming.
     file_content = file.content
-    file_bytes = (
-        file_content.encode("utf-8") if isinstance(file_content, str) else file_content
-    )
-    file_hash = get_sha256(file_content)
 
-    if message_content:
-        if message_content.item_hash != file_hash:
-            raise web.HTTPUnprocessableEntity(
-                reason=f"File hash does not match ({file_hash} != {message_content.item_hash})"
-            )
+    if isinstance(file_content, bytes):
+        file_bytes = file_content
+    elif isinstance(file_content, str):
+        file_bytes = file_content.encode("utf-8")
+    else:
+        raise web.HTTPUnprocessableEntity(reason="Invalid file content type")
```

Comment:
Here, for easier debugging, you want to show the effective file type and the accepted file types